pkg/agent/core/testhelpers/construct.go

```diff
@@ -158,3 +178,15 @@ func WithCurrentCU(cu uint16) VmInfoOpt {
 		vm.SetUsing(c.ComputeUnit.Mul(cu))
 	})
 }
+
+func WithLogicalTime(t *vmapi.LogicalTime) VmInfoOpt {
```

Comment:
Should this be `WithInitialLogicalTime` or something? asking because `WithLogicalTime` IMO makes it seem like the VM only refers to a particular instant in time